use fs.lchown rather than fs.chown and thereby fix #14

simevo opened the pull request:
fixes the symlinks problem #3 while not causing the TOCTOU vulnerability #14

The [patch in libuv 1.21.0](https://github.com/libuv/libuv/releases/tag/v1.21.0) that undeprecates `fs.lchown` [has been incorporated in nodejs Version 10.6.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#2018-07-04-version-1060-current-targos).

So I specified the minimum nodejs version in `package.json` with the `engine` key: https://docs.npmjs.com/files/package.json#engines

Commit:
use fs.lchown rather than fs.chown and thereby fix #14

fixes the symlinks problem #3 while not causing the TOCTOU vulnerability #14

The [patch in libuv 1.21.0](https://github.com/libuv/libuv/releases/tag/v1.21.0) that undeprecates `fs.lchown` [has been incorporated in nodejs Version 10.6.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#2018-07-04-version-1060-current-targos).

So I specified the minimum nodejs version in `package.json` with the `engine` key: https://docs.npmjs.com/files/package.json#engines

afeld commented on package.json:
```diff
@@ -19,5 +19,6 @@
   },
-  "license": "ISC"
+  "license": "ISC",
+  "engines": { "node" : ">=10.6.0" }
```
You'll need to change the `.travis.yml` to match.

afeld replied:
Maybe update that list to match the [maintained Node versions](https://github.com/nodejs/Release#release-schedule)?

simevo replied:
> The job exceeded the maximum time limit for jobs, and has been terminated.

no wonder, it decided to build nodejs 10.6.0 from source !

Commit:
update travis to match the supported nodejs version

see: https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions

in the process:
- update that list to match the maintained Node versions
  see: https://github.com/nodejs/Release#release-schedule
- remove the constraint on the npm version (I dont' see why the default npm would not work)

Commit:
update travis to use a docker image with node 10.8 preinstalled

references:
- https://www.jeffgeerling.com/blog/2018/how-i-test-ansible-configuration-on-7-different-oses-docker
- https://hub.docker.com/_/node/

Commit:
fix docker/travis config

Commit:
docker unknown shorthand flag -w

simevo commented:
Ok now the tests are running on nodejs 10.8 and we get several errors, of 3 types:
- Error: TypeError [ERR_INVALID_ARG_TYPE]: The "gid" argument must be of type number. Received type undefined
- Error: gid should be undefined
- Error: uid not should be 0

the very same 30 tests are failing on master; see: #16

morkro commented:
Since this seems to get no attention from the project maintainer, is there a way to help? 
What guarantees this will even be merged if all checks are successful again?
